### PR TITLE
added global viewport margins for controlling how home bounds make sp…

### LINKED ISSUE #13: Build in configurable margins for viewport to make room for static overlays at the home zoom level.
Components such as Mirador's title bar, bottom panel, and side panel, or a mini-map of the current canvas position or graphic of annotations at different depths would all occlude the content on the canvas in a fully zoomed out detail mode unless their presence is factored into the layout algorithm. This can be accomplished by having configurable margins on the viewport.

### PR DESCRIPTION
**Pull request by aeschylus**
…ace of overlaid element like navigation panels and titles. closes #13

### CODE REVIEW

**Comment by aeschylus** [comment]
@mejackreed 

I'm now trying this out in sul-embed. One question will be how this looks when it is updated. There is currently no method to update the margins dynamically, so perhaps that should be next before pushing to sul-embed.

See the example page for the viewport margin configuration.

File: example/index.html [hunk 1]
```diff
@@ -69,90 +69,96 @@
...
-         }
-
-         $.get('http://iiif.io/api/presentation/2.0/example/fixtures/collection.json', function(fixtureCollection) {
-             var manifestPicker = $('#manifestPicker');
-             fixtureCollection.manifests.forEach(function(manifest) {
-                 var option = $('<option>').val(manifest['@id']).text(manifest.label);
-                 manifestPicker.append(option);
-             });
-         });
+        <script type="text/javascript">
```
Comment (mejackreed) [inline example/index.html:72]: A lot of this seems to be spacing changes, are there other changes in here too?

File: example/index.html [hunk 2]
```diff
@@ -69,90 +69,96 @@
...
+            });
+            key('m', cycleViewingModes);
+
+            var currentMode = 'individuals';
+
+            function cycleViewingModes() {
+            var newMode;
+
+            if (currentMode === 'individuals') {
+            newMode = 'paged';
```
Comment (mejackreed) [inline example/index.html]: What happened to the spacing here?

File: example/index.html [hunk 3]
```diff
@@ -69,90 +69,96 @@
...
+                perspective:  'detail',
+                canvasClass: 'canvas', //default set to 'canvas'
+                frameClass: 'frame', //default set to 'frame'
+                labelClass: 'label', //default set to 'label'
+                viewportPadding: {  // in detail view, make sure this area is clear
+                    top: 0,
+                    left: 10,
+                    right: 10,
+                    bottom: 10 // units in % of pixel height of viewport
+                },
```
Comment (aeschylus) [inline example/index.html]: Yeah(to spacing changes), this section was added. Beyond that, no.

File: example/index.html (hunk 2 above)
Comment (aeschylus) [inline example/index.html]: Good question. I'll tidy this back up. It likely has something to do with my editor's settings.

File: src/manifestLayout.js [hunk 4]
```diff
@@ -55,14 +49,33 @@ var manifestLayout = function(options) {
...
-        aspectRatio: containerWidth/containerHeight
-      };
+      viewport = viewport(containerWidth, containerHeight, options.viewportPadding);
+
+  function viewport(width, height, viewportPadding) {
+    var horizontalMargin,
+        verticalMargin;
+
+    var viewport = {
+      padding: viewportPadding || {
```
Comment (mejackreed) [inline src/manifestLayout.js:59]: Should we validate this at all? eg (top + bottom < 100) .. Just a thought, probably can happen later.

File: example/index.html (hunk 2 above)
Comment (mejackreed) [inline example/index.html]: :+1:

File: example/index.html (hunk 3 above)
Comment (mejackreed) [inline example/index.html]: Cool..

File: src/manifestLayout.js (hunk 4 above)
Comment (aeschylus) [inline src/manifestLayout.js:59]: Yeah, it will be a problem. We should certainly validate it. I'll add that now.

**Comment by aeschylus** [comment]
I feel it would be best to do the checking with try catch and logging errors to the console. Another option is to force them into a reasonable value, though that's a bit invisible. After starting this, I realise this is going to take a little bit of time. I say we merge this and address the validation as a new ticket.

**Comment by aeschylus** [comment]
https://github.com/sul-dlss/iiifManifestLayouts/issues/93

**Comment by mejackreed** [comment]
:+1: sounds good to me

**Pull request merged**